src/realm/integer_compressor.cpp

```diff
@@ -85,6 +87,13 @@ struct IntegerCompressor::VTableForFlex {
 const typename IntegerCompressor::VTableForPacked::PopulatedVTablePacked IntegerCompressor::VTableForPacked::vtable;
 const typename IntegerCompressor::VTableForFlex::PopulatedVTableFlex IntegerCompressor::VTableForFlex::vtable;
 
+using SetDirect = void (*)(char*, size_t, int_fast64_t);
+static std::unordered_map<size_t, SetDirect> s_direct_array_set = {
```

Comment:
Suggestion: It will be faster using a vector of 64xSetDirect where only entries 0,1,2,4,8,16,32,64 is used.
The remaining entries should lead to a SetDirect which asserts so that we can detect wrong use.